motionplan/motionPlanner.go

```diff
@@ -54,10 +54,51 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
```

Comment:
todo: adding a test for this function.
case with 
1. nil frame
2. nil frame system
3. non-nil fs with frame not in fs
4. nil goal
5. non-nil goal but parent of goal not in fs
6. nil start configuration
7. only successcase - nil frame system with goal parent as world